x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts

```diff
@@ -72,6 +73,9 @@ export function initSpacesOnPostAuthRequestInterceptor({
           statusCode: wrappedError.output.statusCode,
         });
       }
+    } else if (isRequestingSpaceRoot) {
+      const destination = addSpaceIdToPath(serverBasePath, spaceId, ENTER_SPACE_PATH);
```

Comment:
We're _technically_ within the context of the requested space here, so we could use `request.getDefaultRoute()` directly if we wanted to (or delegate to the OSS handler). I opted for the redirection to the `ENTER_SPACE_PATH` so that the logic of entering a space was consistent within this interceptor. It's one less code path to have to reason about when interpreting what this interceptor actually does.